lilypad/ee/generate_free_tier_license.py

```diff
@@ -1,77 +1,108 @@
-"""generate_free_tier_license.py
+"""Free-Tier License Key Generator for LilyPad Enterprise Edition
 
-This script demonstrates how to generate a "free-tier" license key
-using the existing generate_license() function from the EE validate module.
+This module generates a free-tier license key for users who wish to self-host
+the system without requiring an externally provided private key. The license is
+signed using an RSA key pair dedicated solely to the free tier. Note that
+the key is not considered sensitive and is only used to enable free-tier
+functionality.
 
 Usage:
-  python generate_free_tier_license.py --private-key /path/to/private_key.pem \
-                                       --password yourKeyPass \
-                                       --customer "Free User" \
-                                       --license-id "FREE-TIER-ABC" \
-                                       --duration-days 365
-
-Notes:
-  - By default, --password is optional. If your key is unencrypted, omit it.
-  - If you omit --customer, it defaults to "FreeTierUser".
-  - The script will print the generated license key to stdout.
+    uv run lilypad/ee/generate_free_tier_license.py --customer "Free User" --license-id "FREE-TIER-ABC" --duration-days 365
 """
 
+import base64
+import json
 from datetime import datetime, timedelta
+from pathlib import Path
+from typing import cast
 
 import typer
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import padding, rsa
 
-from lilypad.ee.validate import LicenseError, generate_license
+app = typer.Typer(
+    help="Generate a free-tier license key for LilyPad Enterprise Edition features."
+)
 
-app = typer.Typer(help="Generate free-tier license keys for LilyPad EE features.")
+FREE_TIER_PRIVATE_KEY_FILE_NAME = "free_tier_private_key.pem"
+FREE_TIER_PRIVATE_KEY = (
+    Path(__file__).parent.joinpath(FREE_TIER_PRIVATE_KEY_FILE_NAME).read_text()
+)
```

Comment:
I considered embedding it as a literal in the Python source code, but decided to externalize it for better usability.